### PR TITLE
Update to a current version of libcxx and libcxxabi

### PR DESCRIPTION
**Pull request by jason-simmons**
This also updates the engine's minimum required version of macOS to 10.13

### CODE REVIEW

**Review by chinmaygarde — APPROVED** [review]

**Comment by chinmaygarde** [comment]
I believe this is good to go. Filed https://github.com/flutter/flutter/issues/107492 to track updating the min deployment target.

File: build/config/mac/mac_sdk.gni [hunk 1]
```diff
@@ -10,7 +10,7 @@ declare_args() {
 
   # The MACOSX_DEPLOYMENT_TARGET variable used when compiling.
   # Must be of the form x.x.x for Info.plist files.
-  mac_deployment_target = "10.11.0"
+  mac_deployment_target = "10.12.0"
```
Comment (jmagman) [inline build/config/mac/mac_sdk.gni:13]: Can you change this to 10.13?  See https://github.com/flutter/flutter/issues/107492#issuecomment-1185054687

**Review by jmagman — CHANGES_REQUESTED** [review]
Requesting changes to make this 10.13

File: build/config/mac/mac_sdk.gni (hunk 1 above)
Comment (jason-simmons) [reply build/config/mac/mac_sdk.gni]: done

Comment (jmagman) [reply build/config/mac/mac_sdk.gni]: Should `mac_sdk_min` also be bumped to 10.13?

Comment (jmagman) [reply build/config/mac/mac_sdk.gni]: I guess it should if everything happily compiles...

Comment (jason-simmons) [reply build/config/mac/mac_sdk.gni]: Updated `mac_sdk_min` to 10.13 and confirmed that the engine builds and runs

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**